DICOM/src/Microsoft.Health.Dicom.Anonymizer.Core/Processors/SubstituteProcessor.cs

```diff
@@ -0,0 +1,73 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Anonymizer.Core.Exceptions;
+using Microsoft.Health.Dicom.Anonymizer.Core.Model;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.Health.Dicom.Anonymizer.Core.Processors
+{
+    public class SubstituteProcessor : IAnonymizerProcessor
+    {
+        private readonly string _replaceWith = "Anonymous";
+
+        public SubstituteProcessor(JObject settingObject)
+        {
+            EnsureArg.IsNotNull(settingObject, nameof(settingObject));
+
+            if (settingObject.TryGetValue("replaceWith", StringComparison.OrdinalIgnoreCase, out JToken replaced))
+            {
+                _replaceWith = replaced.ToString();
+            }
+        }
+
+        public void Process(DicomDataset dicomDataset, DicomItem item, ProcessContext context = null)
+        {
+            EnsureArg.IsNotNull(dicomDataset, nameof(dicomDataset));
+            EnsureArg.IsNotNull(item, nameof(item));
+
+            try
+            {
+                if (item is DicomOtherWord)
```

Comment:
>item is DicomOtherWord [](http://example.com/codeflow?start=20&length=22)

similar question. Why the value type instead of item.ValueRepresentation is used here?

Reply:
item.ValueRepresentation (VR) and Class types (such as DicomOtherWord ) are equivalence (most times) if the Class Type is the leaf node in the derived tree.  
Using VR is better since VR is DICOM standard types and Class types are defined in the library we used. (The relationships is showed in the [graph](https://github.com/sowu880/Dicom-demo/blob/main/docs/classes.vsdx)).
In this processor, VR is better and I will change them.
But for encryption and cryptoHash processors, DicomStringElement derives many classes with different VR. Using class types is more concise.